sycl/doc/extensions/proposed/sycl_ext_intel_fpga_mem.asciidoc

```diff
@@ -0,0 +1,488 @@
+
+  struct port_direction_key {
+    template <port_direction_enum Port_direction>
+    using value_t = property_value<
+        port_direction_key,
+        std::integral_constant<port_direction_enum, Port_direction>>;
+  };
+
+  struct clock_2x_key {
+    template <bool Enable>
+    using value_t = property_value<clock_2x_key, std::bool_constant<Enable>>;
+  };
+
+  enum class ram_stitching_enum : std::uint16_t { min_ram, max_fmax };
+
+  struct ram_stritching_key {
+    template <ram_stritching_enum Ram_stritching>
+    using value_t = property_value<
+        ram_stritching_key,
+        std::integral_constant<ram_stritching_enum, Ram_stritching>>;
+  };
+
+  struct private_copies_key {
+    template <size_t n>
+    using value_t =
+        property_value<private_copies_key, std::integral_constant<n>>;
+  };
+
+  struct num_replicates_key {
+    template <size_t n>
+    using value_t =
+        property_value<num_replicates_key, std::integral_constant<n>>;
+  };
+
+  template <resource_enum r> inline constexpr resource_key::value_t<r> resource;
+  inline constexpr resource_key::value_t<resource_enum::mlab> resource_mlab;
+  inline constexpr resource_key::value_t<resource_enum::block_ram>
+      resource_block_ram;
+
+  template <size_t e> inline constexpr num_banks_key::value_t<e> num_banks;
+
+  template <size_t e> inline constexpr stride_size_key::value_t<e> stride_size;
+
+  template <size_t e> inline constexpr word_size_key::value_t<e> word_size;
+
+  template <port_direction_enum d>
+  inline constexpr port_direction_key::value_t<d> port_direction;
+  inline constexpr port_direction_key::value_t<port_direction_enum::uni>
+      port_direction_uni;
+  inline constexpr port_direction_key::value_t<port_direction_enum::bi>
+      port_direction_bi;
+
+  template <bool b>
+  inline constexpr clock_2x_key::value_t<b> clock_2x;
+  inline constexpr clock_2x_key::value_t<true> clock_2x_true;
+  inline constexpr clock_2x_key::value_t<false> clock_2x_false;
+
+  template <ram_stitching_enum d>
+  inline constexpr ram_stitching_key::value_t<d> ram_stitching;
+  inline constexpr ram_stitching_key::value_t<ram_stitching_enum::min_ram>
+      ram_stitching_min_ram;
+  inline constexpr ram_stitching_key::value_t<ram_stitching_enum::max_fmax>
+      ram_stitching_max_fmax;
+
+  template <size_t n>
+  inline constexpr private_copies_key::value_t<n> private_copies;
+
+  template <size_t n>
+  inline constexpr num_replicates_key::value_t<n> num_replicates;
+
+} // namespace sycl::ext::intel::experimental
+----
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
+
+a|
+[source,c++]
+----
+resource
+----
+a|
+Specifies which FPGA memory resources to use to implement the variable.
+
+The following values are supported:
+
+* `mlab`: data is stored in special Adaptive Logic Modules (ALMs), called memory-logic array blocks 
+* `block_ram`: data is stored in dedicated block RAM modules, called M20K modules
```

Comment:
"M20K" is too Arria 10/Stratix 10/Agilex specific.  Older families have had other memory sizes so it's not impossible that newer families might have different sizes again.